document-store/src/main/java/org/hypertrace/core/documentstore/expression/Filterable.java

```diff
@@ -0,0 +1,7 @@
+package org.hypertrace.core.documentstore.expression;
+
+/**
+ * A marker interface to represent that the expression is filterable and can be used in either the
+ * WHERE clause or the HAVING clause of the query.
+ */
+public interface Filterable {}
```

Comment:
Independent of the other comments about marker interfaces and aggregates, I'm not sure these names are quite right. `Filterable`, in my mind, implies something can be filtered. Here, we're using this to actually say it does the filtering (maybe `Predicate` would be a better fit for this one? Same idea applies to others - more generically, could do things like `FilteringExpression`)

Reply:
Makes sense! Thanks for the naming suggestions. :)